precompiles/werc20/tx.go

```diff
@@ -14,14 +14,15 @@ const (
 
 // Deposit is a no-op and mock function that provides the same interface as the
 // WETH contract to support equality between the native coin and its wrapped
-// ERC-20 (eg. EVMOS and WEVMOS). It only emits the Deposit event.
+// ERC-20 (e.g. EVMOS and WEVMOS).
 func (p Precompile) Deposit() ([]byte, error) {
+	// TODO: Shouldn't this raise an error now? It's a no-op right now, but that invites users to waste gas without anything happening. Wouldn't an error be more clear that this is not useful anymore?
```

Comment:
I think this should not be a no-op but error instead - currently users could still call this and waste gas, even though this functionality does not exist anymore.